controller.go

```diff
@@ -181,6 +258,47 @@ func (c *controller) NewNetwork(networkType, name string, options ...NetworkOpti
 	return network, nil
 }
 
+func (c *controller) newNetworkFromStore(n *network) {
+	c.Lock()
+	n.ctrlr = c
+	n.driver = c.drivers[n.networkType]
+	c.networks[n.id] = n
+	c.Unlock()
+	// TODO : Populate n.endpoints back from endpoint dbstore
+}
+
+func (c *controller) addNetworkToStore(n *network) error {
+	if isReservedNetwork(n.Name()) {
+		return nil
+	}
+	if c.store == nil {
+		return ErrInvalidDatastore
```

Comment:
if c.store is not nil you need to may be just return silently taking this is a non-clustered mode.  Again you need to acquire controller lock